src/vectorsynchronizer.js

```diff
@@ -77,9 +77,10 @@ olcs.VectorSynchronizer.prototype.createSingleCounterpart = function(olLayer) {
   var csPrimitives = olcs.core.olVectorLayerToCesium(olLayer, view,
       featurePrimitiveMap);
 
-  olLayer.on('change:visible', function(e) {
-    csPrimitives.show = olLayer.getVisible();
+  olLayer.on(['change:visible', 'change:opacity'], function(e) {
+    olcs.core.updateCesiumPrimitives(olLayer, csPrimitives);
```

Comment:
Changing visibility should not trigger the expensive modification of all opacities.